src/openrct2/common.h

```diff
@@ -41,14 +43,14 @@ using utf16string = utf16*;
 using codepoint_t = uint32_t;
 using colour_t = uint8_t;
 
-#define rol8(x, shift) (((uint8_t)(x) << (shift)) | ((uint8_t)(x) >> (8 - (shift))))
-#define ror8(x, shift) (((uint8_t)(x) >> (shift)) | ((uint8_t)(x) << (8 - (shift))))
-#define rol16(x, shift) (((uint16_t)(x) << (shift)) | ((uint16_t)(x) >> (16 - (shift))))
-#define ror16(x, shift) (((uint16_t)(x) >> (shift)) | ((uint16_t)(x) << (16 - (shift))))
-#define rol32(x, shift) (((uint32_t)(x) << (shift)) | ((uint32_t)(x) >> (32 - (shift))))
-#define ror32(x, shift) (((uint32_t)(x) >> (shift)) | ((uint32_t)(x) << (32 - (shift))))
-#define rol64(x, shift) (((uint64_t)(x) << (shift)) | ((uint64_t)(x) >> (64 - (shift))))
-#define ror64(x, shift) (((uint64_t)(x) >> (shift)) | ((uint64_t)(x) << (64 - (shift))))
+const constexpr auto rol8 = rol<uint8_t>;
```

Comment:
I think (but I'd like him to confirm it) that @IntelOrca also meant that these consts should be moved.

Reply:
:+1:

Reply:
It doesn't really matter since they are going to be deleted eventually.